### PR TITLE
feat: create waffle switch default_elastic_search

### PR DESCRIPTION
**Pull request by rayzhou-bit**
https://2u-internal.atlassian.net/browse/TNL-9899

This change adds the switch "default_elastic_search". When this switch is on, edx-search will run ElasticSearch if it does not find a search engine. This prevents errors from occuring when an OpenSearch instance cannot be found.

### CODE REVIEW

File: search/search_engine_base.py [hunk 1]
```diff
@@ -58,5 +59,23 @@ def get_search_engine(index=None):
...
+        # .. toggle_target_removal_date: None
+        # .. toggle_tickets: TNL-9899
+        # .. toggle_warnings: This temporary feature toggle does not have a target removal date.
+        DEFAULT_ELASTIC_SEARCH_SWITCH = WaffleSwitch('default_elastic_search', __name__)
+
         search_engine_class = _load_class(getattr(settings, "SEARCH_ENGINE", None), None)
-        return search_engine_class(index=index) if search_engine_class else None
+        if search_engine_class:
+            return search_engine_class(index=index)
+        # TNL-9899
```
Comment (bszabo) [inline search/search_engine_base.py:78]: I believe this TNL-9899 logic should precede the `_load_class()` fetch and return to override existing behavior.

File: search/search_engine_base.py [hunk 2]
```diff
@@ -58,5 +59,23 @@ def get_search_engine(index=None):
...
+        # .. toggle_default: False
+        # .. toggle_description: This flag forces the use of ElasticSearch.
+        #      It prevents errors from switching to OpenSearch before roll out.
+        # .. toggle_use_cases: temporary
+        # .. toggle_creation_date: 2022-7-11
+        # .. toggle_target_removal_date: None
+        # .. toggle_tickets: TNL-9899
+        # .. toggle_warnings: This temporary feature toggle does not have a target removal date.
+        DEFAULT_ELASTIC_SEARCH_SWITCH = WaffleSwitch('default_elastic_search', __name__)
+
```
Comment (bszabo) [inline search/search_engine_base.py:74]: Per comment below,
`if DEFAULT_ELASTIC_SEARCH_SWITCH.is_enabled():
    return None`

**Review by kenclary — APPROVED** [review]
exciting.

**Comment by bszabo** [comment]
I think it would be appropriate to squash the long list of commits.

**Pull request merged**